packages/graph/hash_graph/lib/graph/src/store/mod.rs

```diff
@@ -194,13 +197,17 @@ pub trait Store {
         created_by: AccountId,
     ) -> Result<Persisted<DataType>, InsertionError>;
 
-    /// Get an existing [`DataType`] by a [`VersionId`].
+    /// Get the [`DataType`] specified by `index`.
     ///
     /// # Errors
     ///
     /// - if the requested [`DataType`] doesn't exist.
-    async fn get_data_type(&self, version_id: VersionId)
-    -> Result<Persisted<DataType>, QueryError>;
+    async fn get_data_type<I: Sync>(&self, index: &I) -> Result<Self::Output, QueryError>
+    where
+        Self: Read<I, DataType>,
+    {
+        self.get(index).await
```

Comment:
For my understanding:
The `get_data_type` (etc.) impl for the `PostgresStore` is replaced by the default impl of this trait. Would a `DataTypeStore` trait have the `get_data_type` method in follow ups?

Reply:
Yes, I would stick with `get_data_type` so when implementing multiple stores at the same type, it's possible to distinguish between the names.